### PR TITLE
update runner creation and registration

### PR DESCRIPTION
**Pull request by DaRa42**

### CODE REVIEW

File: {{cookiecutter.project_slug}}/.gitlab-ci.yml [hunk 1]
```diff
@@ -35,13 +35,14 @@ test_{{ cookiecutter.project_slug }}:
         path: coverage.xml
       junit: report.xml
     expire_in: 30 days
-    when: always
+  only:
+    - main
```
Comment (romulogoncalves) [inline {{cookiecutter.project_slug}}/.gitlab-ci.yml]: Why only for the main?

File: {{cookiecutter.project_slug}}/tests/CI_docker/context/{{cookiecutter.project_slug}}_ci.docker [hunk 2]
```diff
@@ -17,13 +16,13 @@ RUN rm Mambaforge-$(uname)-$(uname -m).sh
...
 # Python packages from conda
-RUN mamba install -y python=3.9
+RUN mamba install -y python=3.12
 RUN mamba install -y pip
 RUN mamba init bash
 
 # copy some needed stuff to /root
 COPY *.yml /root/
-RUN mamba env create -n ci_env -f /root/environment_{{ cookiecutter.project_slug }}.yml python=3.10
+RUN mamba env create -n ci_env -f /root/environment_{{ cookiecutter.project_slug }}.yml python=3.12
```
Comment (romulogoncalves) [inline {{cookiecutter.project_slug}}/tests/CI_docker/context/{{cookiecutter.project_slug}}_ci.docker]: I think the option python=3.12 does not work anymore. It was removed for the latest version of mamba.

**Review by romulogoncalves — CHANGES_REQUESTED** [review]

File: {{cookiecutter.project_slug}}/.gitlab-ci.yml (hunk 1 above)
Comment (DaRa42) [reply {{cookiecutter.project_slug}}/.gitlab-ci.yml]: removed that lines

File: {{cookiecutter.project_slug}}/tests/CI_docker/context/{{cookiecutter.project_slug}}_ci.docker (hunk 2 above)
Comment (DaRa42) [reply {{cookiecutter.project_slug}}/tests/CI_docker/context/{{cookiecutter.project_slug}}_ci.docker]: with the changes from @jo-kno this got updated now

File: {{cookiecutter.project_slug}}/.gitlab-ci.yml [hunk 3]
```diff
@@ -35,13 +35,12 @@ test_{{ cookiecutter.project_slug }}:
         path: coverage.xml
       junit: report.xml
     expire_in: 30 days
-    when: always
```
Comment (romulogoncalves) [inline {{cookiecutter.project_slug}}/.gitlab-ci.yml]: Should not we get this line back?

**Review by romulogoncalves — CHANGES_REQUESTED** [review]

File: {{cookiecutter.project_slug}}/.gitlab-ci.yml (hunk 3 above)
Comment (DaRa42) [reply {{cookiecutter.project_slug}}/.gitlab-ci.yml]: right, I put it back in!

**Review by romulogoncalves — APPROVED** [review]

**Pull request merged**